suite-common/connect-popup/src/methodHooks/bitcoinSignTransaction.ts

```diff
@@ -0,0 +1,105 @@
+import { Bip43Path, NetworkSymbol, getNetwork } from '@suite-common/wallet-config';
+import {
+    accountsActions,
+    selectAccountForNetworkSymbolAndPath,
+    sendFormActions,
+} from '@suite-common/wallet-core';
+import { Account, FormOptions } from '@suite-common/wallet-types';
+import TrezorConnect from '@trezor/connect';
+import type { SignTransaction } from '@trezor/connect';
+import { getSerializedPath } from '@trezor/connect/src/utils/pathUtils';
+
+import { connectPopupActions } from '../connectPopupActions';
+import { createPlaceholderAccount } from './utils';
+
+import { PostCallHookParams, PreCallHookParams } from './index';
+
+const temporaryAccounts: Account[] = [];
+
+const preCallHook = async <M extends keyof typeof TrezorConnect>({
+    method,
+    payload,
+    getState,
+    dispatch,
+    txSigningPrecomposed,
+}: PreCallHookParams<M>) => {
+    try {
+        if (method === 'signTransaction' && txSigningPrecomposed) {
+            const typedPayload = payload as any as SignTransaction;
```

Comment:
I didn't test it yet, but shouldn't it the payload object be cloned here? see similar commit here https://github.com/trezor/trezor-suite/pull/18322/commits/fd5339b05adde5c8ab775d5b41272d0be828d8cb

Reply:
We're not modifying the payload in this case, just reading from it, so it shouldn't be needed